packages/core/src/app/block-production/merkle-tree.ts

```diff
@@ -32,16 +32,22 @@ export class SparseMerkleTreeImpl implements SparseMerkleTree {
   private readonly treeLock: AsyncLock = new AsyncLock({
     domainReentrant: true,
   })
+  private readonly hashFunction: (Buffer) => Buffer
   private readonly hashBuffer: Buffer = Buffer.alloc(64)
 
   constructor(
     private readonly db: DB,
     rootHash?: Buffer,
     private readonly height: number = 160,
```

Comment:
Is the tree height still kind of "off by 1" in that an inclusion proof for a depth 32 tree has 31 siblings? If so we should update it (doesn't have to be internally, but deff from the contract perspective) to be depth 32 has 32 siblings

Reply:
@karlfloersch I thought we agreed that the root counts in height. The root never has a sibling, so sibling count will always be depth - 1.

Reply:
yeah I was into the root counting as the height, but after using a bunch of merkle tree libraries it looks like it's super standard practice to say a tree of depth 1 has a root and some children.

if it's not terribly troubling do you think we can adjust to make the height == number of siblings? We can also leave as is tbh or I can change in the contracts, but seeing as it's non-standard i prefer either doing nothing for now or changing it here

Reply:
Just noting this for the record. As we discussed, yes, we should do this, but we'll wait until after Devcon. I'll created #472 to address it.